src/main/java/com/opentable/versionedconfig/GitOperations.java

```diff
@@ -108,11 +108,17 @@ boolean pull() throws VersioningServiceException {
         LOG.trace("pulling latest");
         return upstreamRetry(remoteIndex -> {
             try {
+                int timeout = config.getTimeoutInSec();
                 final PullCommand pull = git.pull();
                 LOG.trace("Git pull completed");
                 configureCredentials(pull, config.getRemoteRepositories().get(remoteIndex));
                 LOG.trace("Configuration of credentials completed, setting remote {}", remoteIndex);
                 pull.setRemote("remote" + remoteIndex);
+
+                //set timeout if defined
+                if (timeout != -1){
```

Comment:
This assumes 0 does what you want, which is likely but bad coding practice. Probably it should be > 0